api/handler/v1beta1/job_spec.go

```diff
@@ -285,11 +286,45 @@ func (sv *JobSpecServiceServer) DeleteJobSpecification(ctx context.Context, req
 	}, nil
 }
 
-func NewJobSpecServiceServer(l log.Logger, jobService models.JobService, adapter ProtoAdapter, namespaceService service.NamespaceService, progressObserver progress.Observer) *JobSpecServiceServer {
+func (sv *JobSpecServiceServer) RefreshJobs(req *pb.RefreshJobsRequest, respStream pb.JobSpecificationService_RefreshJobsServer) error {
+	startTime := time.Now()
+
+	observers := new(progress.ObserverChain)
+	observers.Join(sv.progressObserver)
+	observers.Join(&jobRefreshObserver{
+		stream: respStream,
+		log:    sv.l,
+		mu:     new(sync.Mutex),
+	})
+
+	namespaceJobNamePairs := sv.prepareNamespaceJobNamePairs(req.NamespaceJobs)
+
+	if err := sv.jobSvc.Refresh(respStream.Context(), req.ProjectName, namespaceJobNamePairs, observers); err != nil {
```

Comment:
can you split this into multiple lines, will be more readable

Reply:
Can you change the proto request to accept request for project refresh or bunch of namespaces in project or bunch of jobs. project is mandatory. namespaces and jobs list is optional.
We can have different functions for refreshing all jobs for a project. refresh all jobs of given namespaces, refresh bunch of jobs in a project.

Reply:
is this what you meant?
```proto
message RefreshJobsRequest {
  string project_name = 1;
  repeated string namespace_names = 2;
  repeated string job_names = 3;
}
```
currently, to refresh a bunch of jobs in a project, we need namespace info of where the jobs belong (for auth). we can go with what you proposed, but will still require the namespace that needs to be provided in the `namespace_names` (which is not straightforward), or we can avoid requesting the namespace for this case.

Reply:
updated, the proto PR: https://github.com/odpf/proton/pull/127